tokio-util/src/time/mod.rs

```diff
@@ -16,6 +16,33 @@ pub mod delay_queue;
 
 #[doc(inline)]
 pub use delay_queue::DelayQueue;
+use futures_core::Future;
+use tokio::time::Timeout;
+
+/// A trait which contains a variety of conevenient adapters and utilities for `Future`s.
+pub trait FutureExt: Future + Sized {
+    /// A wrapper around [`tokio::time::timeout`], with the advantage that it is easier to write
+    /// fluent call chains.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use tokio::{sync::oneshot, time::Duration};
+    /// use tokio_util::time::FutureExt;
+    ///
+    /// # async fn dox() {
+    /// let (tx, rx) = oneshot::channel::<()>();
+    ///
+    /// let res = rx.timeout(Duration::from_millis(10)).await;
+    /// assert!(res.is_err());
+    /// # }
+    /// ```
+    fn timeout(self, timeout: Duration) -> Timeout<Self> {
+        tokio::time::timeout(timeout, self)
+    }
```

Comment:
Please use this mechanism instead of specifying `Sized` as a sub-trait.
```suggestion
    fn timeout(self, timeout: Duration) -> Timeout<Self>
    where
        Self: Sized,
    {
        tokio::time::timeout(timeout, self)
    }
```

Reply:
Done, thanks